src/composables/accounts.ts

```diff
@@ -115,47 +169,74 @@ export function useAccounts({ store }: IDefaultComposableOptions) {
     }
   }
 
+  function setActiveAccountByProtocolAndIdx(protocol: Protocol, idx: number) {
+    const accountFound = accountsGroupedByProtocol.value[protocol]
+      ?.find((account) => account.idx === idx);
+    if (accountFound) {
+      setActiveAccountByGlobalIdx(accountFound.globalIdx);
+    }
+  }
+
+  function setMnemonic(val: string) {
+    mnemonic.value = val;
+  }
+
+  function setGeneratedMnemonic() {
+    setMnemonic(generateMnemonic());
+  }
+
   /**
    * Determine if provided address belongs to any of the current user's accounts.
    */
   function isLocalAccountAddress(address: Encoded.AccountAddress): boolean {
     return accountsAddressList.value.includes(address);
   }
 
-  function incrementProtocolNextAccountIdx(protocol: Protocol) {
-    if (protocolNextAccountIdx.value[protocol] === undefined) {
-      protocolNextAccountIdx.value[protocol] = 0;
-    } else {
-      protocolNextAccountIdx.value[protocol]! += 1;
-    }
-  }
-
-  function setProtocolNextAccountIdx(value: number, protocol: Protocol) {
-    protocolNextAccountIdx.value[protocol] = value;
+  function addRawAccount({
+    isRestored,
+    protocol,
+  }: Omit<IAccountRaw, 'type'>): number {
+    accountsRaw.value.push({
+      protocol,
+      isRestored,
+      type: ACCOUNT_HD_WALLET,
+    });
+    const lastProtocolAccount = getLastProtocolAccount(protocol);
+    const idx: number = (lastProtocolAccount) ? lastProtocolAccount.idx : 0;
+    return idx;
```

Comment:
I think that assignment is not needed here
```suggestion
    return (lastProtocolAccount) ? lastProtocolAccount.idx : 0;
```